managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerImpl.java

```diff
@@ -2547,21 +2582,15 @@ void internalTrimLedgers(boolean isTruncate, CompletableFuture<?> promise) {
             store.asyncUpdateLedgerIds(name, getManagedLedgerInfo(), ledgersStat, new MetaStoreCallback<Void>() {
                 @Override
                 public void operationComplete(Void result, Stat stat) {
+                    // perform actual deletion
+                    removeAllDeletableLedgers();
```

Comment:
it will block at most 30s to wait all the ledger delete operation completed. However, this method is called by meta store callback thread, which will block the thread and lead to deadlock.

Reply:
The reason for performming `removeAllDeletableLedgers` synchronously here is that two locks of `trimmer` and `metadata` need to be released after the actual deletion. If the execution is asynchronous, the update process of the property map cannot be protected.

Reply:
The operationComplete will be called by `bookkeeper-ml-scheduler` thread pool, If it executed by synchronously, it will block other operations. We can use another new thread pool the execute it. 
For the property map problem, could we use another thread safe data structure the store it?